.github/workflows/release.yml

```diff
@@ -0,0 +1,48 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      pypi:
+        type: choice
+        description: Choose which PyPI to publish to
+        options:
+        - main
+        - test
+        default: main
+
+permissions: read-all
+
+defaults:
+  run:
+    shell: bash
+
+# will cancel previous workflows triggered by the same event and for the same ref for PRs or same SHA otherwise
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ contains(github.event_name, 'pull_request') && github.event.pull_request.head.ref || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    name: ${{ inputs.pypi }}
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Setup environment
+        uses: ./.github/actions/setup-environment
+        with:
+          python-version: "3.8"
```

Comment:
Can we use anything later than 3.8 please.

Reply:
I updated to py311